knowledge_base/RabbitMQ.md

```diff
@@ -1,147 +1,112 @@
+As part of our hexagonal ports-and-adapters architecture, we define interfaces for external services. For message brokers specifically (including, but not limited to RabbitMQ) we use a [`Broker` port](../libs/core/src/ports/interfaces.ts)—a generic publish-subscribe-initialize interface.
 
 ## Rascal
 
-In order to manage RabbitMQ configuration and connections we use [Rascal](https://www.npmjs.com/package/rascal). The current Rascal/RabbitMQ configuration can be found [here](../libs/adapters/src/rabbitmq/rabbitMQConfig.ts).
-
-## Management Panel
-
-### Access
-
-1. See the [FAQ](#how-do-i-access-the-correct-rabbitmq-management-dashboard) for instructions on getting your RabbitMQ Management panel URL.
-
-2. If you are running a local instance of RabbitMQ the username is 'guest' and the password is 'guest'.
-
-### Queues
-
-The Queues tab shows a list of all the existing queues. Columns:
-
-- State
-  - Will usually vary between 'idle' and 'ready'. Either of these values is completely normal. If state is neither of these then contact an admin immediately.
-- Ready
-  - The number of messages that are ready to be taken out of the queue by a consumer.
-- Unacked
-  - The number of messages that consumers have taken from the queue but not acknowledged. This indicates a consumer is busy processing messages or is stuck and is not acking messages. A large number of unacked messages may indicate there is a problem.
-- Total
-  - Ready + Unacked
+To manage RabbitMQ configuration and connections we use [Rascal](https://www.npmjs.com/package/rascal). Rascal publications and subscriptions are abstractions over RabbitMQ exchanges' & queues' inputs and outputs, respectively. The current Rascal/RabbitMQ configuration can be found [here](../libs/adapters/src/rabbitmq/rabbitMQConfig.ts).
 
-If you click on a specific queue there are several things you can do:
-
-- Get a message
-  - You can pull a message from the queue. Depending on the option that you pick the message be or may not be re-queued.
-- Delete Queue
-  - You can delete the queue and all of its messages
-- Purge Messages
-  - You can empty a queue out without deleting it.
-- Publish Messages
-  - You can publish messages to the queue.
+## Policies & error handling
 
-## Monitoring
+When subscribing to Rascal, you must provide a policy. A policy is a strictly typed object that encodes functions to handle various message types. As part of that policy, you can find a retry strategy for that policy when you subscribe.
 
-### Locally
+A retry strategy is defined by the `buildRetryStrategy` function. Strategies include:
 
-Access the management panel of the RabbitMQ instance you are using. See the [FAQ](#how-do-i-access-the-correct-rabbitmq-management-dashboard) for instructions.
+- `deadLetterQueue`: Houses all unprocessable events
```

Comment:
@timolegros Flagging this section for closer review, because it's the part of your explanation I struggled most to understand!

Reply:
The 3 strategies listed exist by default even if you use the `buildRetryStrategy` function. These 3 retry strategies are only unavailable if you define a RetryStrategy for a policy without the buildRetryStrategy function. Most policies will not require use of the `buildRetryStrategy` function as the defaults will suffice.